app/src/main/java/com/eveningoutpost/dexdrip/WidgetUpdateService.java

```diff
@@ -55,61 +35,41 @@ public WidgetUpdateService() {}
 
     @Override
     public void onCreate() {
+        context = getApplicationContext();
         super.onCreate();
-        PowerManager pm = (PowerManager) getSystemService(Service.POWER_SERVICE);
-        Log.d(TAG, "onCreate");
-        if ((Build.VERSION.SDK_INT >= Build.VERSION_CODES.LOLLIPOP && pm.isInteractive()) ||
-                (Build.VERSION.SDK_INT < Build.VERSION_CODES.LOLLIPOP && pm.isScreenOn()))
-            enableClockTicks();
-        else
-            disableClockTicks();
     }
 
-    private void enableClockTicks() {
+    @Override
+    public int onStartCommand(Intent intent, int flags, int startId) {
+        //Gear widget needs clock ticks all the time to keep time updated in widget
         Log.d(TAG, "enableClockTicks");
         IntentFilter intentFilter = new IntentFilter();
         intentFilter.addAction(Intent.ACTION_TIME_TICK);
-        intentFilter.addAction(Intent.ACTION_SCREEN_ON);
```

Comment:
It looks from the diff like this will always be listening for clock ticks. This isn't good for anyone who isn't using wearable widgets from a power saving point of view so I would suggest that the original behavior is restored but with it always enabling listening for clock ticks when the wearable widgets preference item I suggested is enabled. I can add that item in today so you have something to work with easily.